packages/angular/src/generators/component/lib/normalize-options.ts

```diff
@@ -1,29 +1,33 @@
-import type { Tree } from '@nrwl/devkit';
+import type { ProjectConfiguration, Tree } from '@nrwl/devkit';
 import { joinPathFragments, readProjectConfiguration } from '@nrwl/devkit';
+import { parseName } from '../../utils/names';
+import { buildSelector } from '../../utils/selector';
 import type { NormalizedSchema, Schema } from '../schema';
 
-export async function normalizeOptions(
+export function normalizeOptions(
   tree: Tree,
   options: Schema
-): Promise<NormalizedSchema> {
-  const { projectType, root, sourceRoot } = readProjectConfiguration(
+): NormalizedSchema {
+  const { prefix, projectType, root, sourceRoot } = readProjectConfiguration(
     tree,
     options.project
-  );
-  const projectSourceRoot = sourceRoot ?? joinPathFragments(root, 'src');
+  ) as ProjectConfiguration & { prefix?: string };
 
-  const parsedName = options.name.split('/');
-  const name = parsedName.pop();
-  const namedPath = parsedName.join('/');
+  const projectSourceRoot = sourceRoot ?? joinPathFragments(root, 'src');
+  const { name, path: namePath } = parseName(options.name);
```

Comment:
Sorry, I know this function was added in a previous PR, but looking back it here, what do you think of renaming it to this?

It can be done in a follow-up PR.

```suggestion
  const { name, path: namePath } = parseNameWithPath(options.name);
```